src/pretix/base/metrics.py

```
@@ -0,0 +1,150 @@
+        """
+        if not labelset:
+            return metricname
+        else:
+            metricname += "{"
+            for labelname in self.labelnames:
+                metricname += '{0}="{1}",'.format(labelname, labelset[labelname])
+
+            return metricname[:-1] + "}"
+
+    def _inc_in_redis(self, key, amount):
+        """
+        Increments given key in Redis.
+        """
+        rkey = ns_prefix + key  # effective key for redis
+        r.incrby(rkey, amount)
+
+    def _set_in_redis(self, key, value):
+        """
+        Sets given key in Redis.
+        """
+        rkey = ns_prefix + key  # effective key for redis
+        r.set(rkey, value)
+
+
+class Counter(Metric):
+    """
+    Counter Metric Object
+    Counters can only be increased, they can neither be set to a specific value
+    nor decreased.
+    """
+
+    def inc(self, amount=1, **kwargs):
+        """
+        Increments Counter by given amount for the labelset specified in kwargs.
+        """
+        if amount < 0:
+            raise MetricsError("Counter cannot be decreased.")
+
+        self._check_label_consistency(kwargs)
+
+        fullmetric = self._construct_metric_identifier(self.name, kwargs)
+        self._inc_in_redis(fullmetric, amount)
+
+
+class Gauge(Metric):
+    """
+    Gauge Metric Object
+    Gauges can be set to a specific value, increased and decreased.
+    """
+
+    def set(self, value=1, **kwargs):
+        """
+        Sets Gauge to a specific value for the labelset specified in kwargs.
+        """
+        self._check_label_consistency(kwargs)
+
+        fullmetric = self._construct_metric_identifier(self.name, kwargs)
+        self._set_in_redis(fullmetric, value)
+
+    def inc(self, amount=1, **kwargs):
+        """
+        Increments Gauge by given amount for the labelset specified in kwargs.
+        """
+        if amount < 0:
+            raise MetricsError("Amount must be greater than zero. Otherwise use dec().")
```

I disagree, increase and decrease are different operations. Besides that I don't think silently using one of them is a good idea, if I remember the spec correctly I don't think that this is allowed.